lib/repl.js

```diff
@@ -386,14 +386,15 @@ function REPLServer(prompt,
   self.bufferedCommand = '';
   self.lines.level = [];
 
-  function complete(text, callback) {
-    self.complete(text, callback);
-  }
+  // Figure out which "complete" function to use.
+  self.completer = (typeof options.completer === 'function')
+    ? options.completer
+    : complete.bind(self);
```

Comment:
I see what you mean. Could you add a test that fails when it is not bound? I just pulled down your changes, removed the `bind()`, and the entire test suite still passed.